keps/sig-release/20190118-creating-packages.md

```diff
@@ -0,0 +1,417 @@
+  - *Description*:  
+    These tools are part of the official release process. After a new
+    kubernetes version has been cut and released, but before the new release
+    was announced, these tools will pull the build binaries from a bucket and
+    wrap them in a `deb`/`rpm`.
+  - *Usage*:
+    ```
+    k/release # ./debian/jenkins.sh
+    k/release # cd ./rpm/ && ./docker-build.sh
+    ```
+  - *Cons*<a id="k-release-cons"></a>:
+    - It uses a different (no) branching strategy then the artefacts build from `k/k` it is packaging
+    - Pulls pre-existing artefacts from a GCS bucket
+    - Cannot build just one package, builds all the packages for all architectures (on linux)
+    - Needs manual change of the source code when bumping versions
+  - *Pros*:
+    - builds for multiple platforms (`linux/aarch64`, `linux/armhfp`, `linux/ppc64le`, `linux/s390x`, `linux/x86_64`)
+    - also handles the upload of the packages (at least for `deb`s)
+      - might also be considered a [con](#k-release-cons)?
+  - *Parts to reuse*:
+    - multi-arch support
+- [`k/k/debs`][k-k-deb] & [`k/k/rpms`][k-k-rpm]
+  - *Description*:  
+    This is building the packages directly with bazel. Those packages are not
+    used as official releases -- usecase is not clear.
+  - *Usage*:
+    ```
+    k/k # bazel build //build/debs
+    k/k # bazel build //build/rpms
+    ```
+  - *Cons*:
+    - No multi-arch support, builds only `linux/amd64`
+    - rpm packaging needs `rpmbuild` to be installed on the host
+  - *Pros*:
+    - `bazel` integration
+    - picks up proper version from git
+    - package specs and additional files (systemd units, ...) are versioned in 
+  - *Parts to reuse*:
+    - some sort of `bazel` integration
+    - versioning of the packaging specs and additional files alongside the code
+
+[k-release]: https://github.com/kubernetes/release/tree/master/
+[k-k]: https://github.com/kubernetes/kubernetes/tree/master/build/
+[k-release-deb]: https://github.com/kubernetes/release/tree/master/debian
+[k-release-rpm]: https://github.com/kubernetes/release/tree/master/rpm
+[k-k-deb]: https://github.com/kubernetes/kubernetes/tree/master/build/debs
+[k-k-rpm]: https://github.com/kubernetes/kubernetes/tree/master/build/rpms
+
+To a good part this KEP tries to address most the [issue "k/k is the canonical source of all build artefacts"](https://github.com/kubernetes/kubernetes/issues/71677) and related
+issues:
+- [`kubeadm` package should depend on the same version of `kubelet` and
+  `kubectl` packages](https://github.com/kubernetes/kubernetes/issues/72871)
+- [Automate deb/rpm publishing](https://github.com/kubernetes/sig-release/issues/10)
+- ...
+
+
+
+### Goals
+
+- Fold the packaging specs, systemd unit files, dependency configuration, and
+  "Parts to reuse" into one place in `k/k`.
+
+  As a first step, the state from [`k/release/{rpm,debian}`][k-release] should
+  be used, as those are the things we release right now. As a second step all
```

Comment:
yes. this is the currently used tools and we should start from here for consistency sake,